pentaho-js/src/test/javascript/cdf/components/XactionComponent-spec.js

```diff
@@ -56,5 +60,22 @@ define([
 
       dashboard.update(xactionComponent);
     });
+
+    it("Check if update calls getCdfXaction method with correct parameters", function(done) {
```

Comment:
First, ideally, the title of a test case _should_ start with "should".
Something like "should call getCdfXAction method with correct parameters".

Then, the test title _should_ describe what is effectively being tested. In this case, you are testing one of the cases, where "`path` ends with `action` and `action` is preceded by `/`".

So, a test title like the following would be more appropriate: "should call `getCdfXAction` with an empty action, when `path` ends with `action` preceded by `/`".

Lastly, you are not testing other main case, in which `path` does _not_ end with `action`. I think you should exercise the following (some, corner'y) cases:
- `action` is `null`
- `action` is `undefined`
- `action` is ""
- `path` equals `action`,
- `path` is shorter than `action`
- `path` is longer but does not end with `action`
- `path` ends with `action` and `action` is preceded by `/`
- `path` ends with `action` and `action` is preceded by `\`
- `path` ends with `action` but `action` is preceded by something other than `/` or `\`

To be able to test these different cases, you should refactor the tests to not create the dashboard directly within the body of the `describe` block (which is always bad practice anyway, unless you're declaring a constant...).

You can define a function which creates a dashboard, given a path argument. Then, call it within the individual test cases (i.e. `it` functions).